### PR TITLE
fix: handle undefined updated_at on messages with util function

### PR DESCRIPTION
**Pull request by Enigma-I-am**
## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) spec
- [ ] New code is covered by tests
- [ ] Screenshots added for visual changes
- [ ] Documentation is updated

### CODE REVIEW

File: package/src/utils/utils.ts [hunk 1]
```diff
@@ -512,3 +513,16 @@ export const emojiRegex =
 
 export const urlRegex =
   /(?:\s|^)((?:https?:\/\/)?(?:[a-z0-9-]+(?:\.[a-z0-9-]+)+)(?::[0-9]+)?(?:\/(?:[^\s]+)?)?)/g;
+
+export const mapMessages = <
```
Comment (santhoshvai) [inline package/src/utils/utils.ts:517]: `mapMessages` is such a generic naming. It doesn't document what this function does well. 

The returned string is used as a dependency for the useMemo. This must be clear. Add this is a documentation to the function. 

And also please think of a better name for the function. `mapMessages` => `getUpdatedMessages`, or something similar

File: package/src/components/Channel/hooks/useCreatePaginatedMessageListContext.ts [hunk 2]
```diff
@@ -17,15 +18,9 @@ export const useCreatePaginatedMessageListContext = <
...
-    .map(
-      ({ deleted_at, latest_reactions, reply_count, status, updated_at }) =>
-        `${deleted_at}${
-          latest_reactions ? latest_reactions.map(({ type }) => type).join() : ''
-        }${reply_count}${status}${updated_at?.toISOString?.() || updated_at}`,
-    )
-    .join();
+  }) => {
+  
+  const messagesUpdated = messages.map((props) => mapMessages(props)).join();
```
Comment (santhoshvai) [inline package/src/components/Channel/hooks/useCreatePaginatedMessageListContext.ts:22]: `messages.map(mapMessages).join();` this should be enough

**Review by santhoshvai — CHANGES_REQUESTED** [review]
Nice move to a common util function 👍  Just requesting changes to the naming of it

File: package/src/utils/utils.ts [hunk 3]
```diff
@@ -512,3 +513,16 @@ export const emojiRegex =
 
 export const urlRegex =
   /(?:\s|^)((?:https?:\/\/)?(?:[a-z0-9-]+(?:\.[a-z0-9-]+)+)(?::[0-9]+)?(?:\/(?:[^\s]+)?)?)/g;
+
+export const getMessagesComparatorCallback = <
```
Comment (santhoshvai) [inline package/src/utils/utils.ts:517]: Can you remove `callback` from the name? This function will not be used as a callback.

Also better to add some documentation here.

**Review by santhoshvai — CHANGES_REQUESTED** [review]
Just requesting minimal changes again. Documentation is important.

File: package/src/utils/utils.ts (hunk 3 above)
Comment (vishalnarkhede) [reply package/src/utils/utils.ts:517]: How about simply a `compareMessages`?

Comment (Enigma-I-am) [reply package/src/utils/utils.ts:517]: ✅

File: package/src/utils/utils.ts (hunk 1 above)
Comment (Enigma-I-am) [reply package/src/utils/utils.ts]: ✅

File: package/src/components/Channel/hooks/useCreatePaginatedMessageListContext.ts (hunk 2 above)
Comment (Enigma-I-am) [reply package/src/components/Channel/hooks/useCreatePaginatedMessageListContext.ts]: ✅

File: package/src/utils/utils.ts (hunk 3 above)
Comment (Enigma-I-am) [reply package/src/utils/utils.ts:517]: ✅

**Review by madsroskar — APPROVED** [review]

File: package/src/components/Channel/hooks/useCreatePaginatedMessageListContext.ts [hunk 4]
```diff
@@ -18,14 +19,7 @@ export const useCreatePaginatedMessageListContext = <
...
 }) => {
-  const messagesUpdated = messages
-    .map(
-      ({ deleted_at, latest_reactions, reply_count, status, updated_at }) =>
-        `${deleted_at}${
-          latest_reactions ? latest_reactions.map(({ type }) => type).join() : ''
-        }${reply_count}${status}${updated_at?.toISOString?.() || updated_at}`,
-    )
-    .join();
+  const messagesUpdated = messages.map(compareMessages).join();
```
Comment (santhoshvai) [inline package/src/components/Channel/hooks/useCreatePaginatedMessageListContext.ts:22]:
```suggestion
  const messagesUpdated = compareMessages(messages)
```

**Comment by Enigma-I-am** [comment]
Dicussion with @vishalnarkhede we agreed upon following changes
- rename constants` messagesUpdated `& `threadMessagesUpdated `=> `messagesStr` and `threadMessagesStr` respectively
- rename `compareMessages` and `mapMessage` functiona to `reduceMessagesToString` and `stringifyMessage`
- update inline doc.

File: package/src/utils/utils.ts [hunk 5]
```diff
@@ -512,3 +513,32 @@ export const emojiRegex =
...
+  reply_count,
+  status,
+  updated_at,
+}: FormatMessageResponse<StreamChatGenerics>): string =>
+  `${deleted_at}${
+    latest_reactions ? latest_reactions.map(({ type }) => type).join() : ''
+  }${reply_count}${status}${updated_at?.toISOString?.() || updated_at}`;
+
+/**
+ * reduces a list of messages to strings that are used in useEffect & useMemo
```
Comment (santhoshvai) [inline package/src/utils/utils.ts:536]:
```suggestion
 * Reduces a list of messages to strings that are used in useEffect & useMemo
```
Nitpick

**Review by santhoshvai — APPROVED** [review]

File: package/src/utils/utils.ts (hunk 5 above)
Comment (Enigma-I-am) [reply package/src/utils/utils.ts]: ✅

**Pull request merged**